mb-solrquerywriter/src/main/java/org/musicbrainz/search/solrwriter/MBXMLWriter.java

```diff
@@ -421,20 +448,71 @@ public void write(Writer writer, SolrQueryRequest req, SolrQueryResponse res)
 			xmlList.add(unmarshalledObj);
 		}
 
-		XMLGregorianCalendar now = getNow();
+	}
 
-		Metadata metadata = metadatalistwrapper.getCompletedMetadata();
-		metadata.setCreated(now);
 
-		try {
-			marshaller.marshal(metadata, sw);
-		} catch (JAXBException e) {
-			e.printStackTrace();
-			return;
+	public void parseSolrResponse(SolrDocumentList doclist, MetadataListWrapper metadatalistwrapper)
+	{
+
+
+		metadatalistwrapper.setCountAndOffset(doclist.getNumFound(),
+				doclist.getStart());
+
+		List xmlList = metadatalistwrapper.getLiveList();
+
+		float maxScore = doclist.getMaxScore();
+		Iterator<SolrDocument> iter = doclist.iterator();
+
+		while (iter.hasNext()) {
+			String store;
+			SolrDocument doc = iter.next();
+			try {
+				store = (String) doc.get("_store");
+			} catch (NullPointerException e) {
+				throw new RuntimeException(
+						NO_STORE_VALUE);
+			}
+			if (store == null) {
+				throw new RuntimeException(
+						STORE_NOT_A_STRING);
+			}
+			Object unmarshalledObj;
+			try {
+				unmarshalledObj = unmarshaller
+						.unmarshal(new ByteArrayInputStream(store.getBytes()));
+			} catch (JAXBException e) {
+				// Propagate the error to the user. By simply repeating the same search without mbxml/mbjson response
+				// writer, we can figure out which document caused this.
+				throw new RuntimeException(UNMARSHALLING_STORE_FAILED + store);
```

Comment:
Not related to this pull request, but I think this is one exception that appears to show up in the logs (unfortunately) somewhat regularly, as you mentioned on IRC and it doesn't contain enough information. Javas exceptions usually have constructors that take an additional parameter - the original exception that caused them (like `raise ... from` in Python) and so does `RuntimeException`: https://docs.oracle.com/javase/7/docs/api/java/lang/RuntimeException.html#RuntimeException(java.lang.String,%20java.lang.Throwable). Maybe by using that the logs will contain more information.